[#1793] Initial work making media-view an editor

cadecairos opened the pull request.

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/1bb3ed92c434e1a/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/1bb3ed92c434e1a/output.txt

Total script time: 0.29 mins

Commit:
[#1793] Media Editor

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/4c6b59afa0b2de4/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/4c6b59afa0b2de4/output.txt

Total script time: 0.55 mins

Server is running on http://pr715.butyr.org/templates/test.html

cadecairos commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/61ac3506a3ab876/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/61ac3506a3ab876/output.txt

Total script time: 0.42 mins

secretrobotron commented on src/ui/ui.js:
```diff
@@ -4,10 +4,10 @@
 define( [ "core/eventmanager", "./toggler",
           "./header", "./unload-dialog",
-          "./tray" ],
+          "./tray", "core/views/media-view" ],
```
Do you mind renaming all of these files and variables to `-editor` and `___Editor` respectively? It's not a view anymore, right? It's very much an editor.

cadecairos replied:
good call. should I move the file out of src/core/views too?

ScottDowne commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/49ac4cd843cf4ea/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/49ac4cd843cf4ea/output.txt

Total script time: 0.30 mins

ScottDowne commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/22555bfeba55eb2/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/22555bfeba55eb2/output.txt

Total script time: 0.81 mins

Server is running on http://pr715.butyr.org/templates/basic/

Commit:
[#1793] Fix up Media Editor styles

cadecairos commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/ae8d5c8bab90c88/output.txt

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/9929643ee27100b/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/ae8d5c8bab90c88/output.txt

Total script time: 2.00 mins

ghost commented:
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/9929643ee27100b/output.txt

Total script time: 2.00 mins

Commit:
[#1793] Fix typo

Commit:
[#1793] Fix title inside of media-editor less file

cadecairos commented:
/botio preview

Commit:
[#1793] Add limit to total current media. Cleaned up code a bit.

Commit:
[#1793]
1. Update UI
2. Bug fixes
3. Remembers empty url boxes
4. 1000% more epic

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/fec2b1bf72d0bf6/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/fec2b1bf72d0bf6/output.txt

Total script time: 0.46 mins

Server is running on http://pr715.butyr.org/templates/basic/
Run the Test Suite: http://pr715.butyr.org/test/

Commit:
[#1793]
1. Primary media cannot be deleted
2. Added hover over styles for buttons
3. Fixed styling of alternate media inputs
4. Moved error label below alternate source button
5. Added bottom margin to primary media wrapper div

cadecairos commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/f2b47edb108ff89/output.txt

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/a9f7ccaed69e0a5/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/f2b47edb108ff89/output.txt

Total script time: 1.75 mins

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/a9f7ccaed69e0a5/output.txt

Total script time: 1.14 mins

Server is running on http://pr715.butyr.org/templates/basic/
Run the Test Suite: http://pr715.butyr.org/test/

mjschranz commented on src/editor/media-editor.js:
```diff
@@ -2,20 +2,272 @@
+          inputs[ i ].classList.add( "error" );
+        } else {
+          inputs[ i ].classList.remove( "error" );
+        }
```
Nit picks, but you could just write something like....

<pre>
inputs[ i ].classList.toggle( "error" );
</pre>


And forego the if statement checking state

mjschranz commented on src/editor/media-editor.js:
```diff
@@ -2,20 +2,272 @@
+        _mediaErrorMessage.classList.remove( "hidden" );
+      } else {
+        _mediaErrorMessage.classList.add( "hidden" );
+      }
```
Same as above.

mjschranz commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/330e0e3241bfbde/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/330e0e3241bfbde/output.txt

Total script time: 0.74 mins

Server is running on http://pr715.butyr.org/templates/basic/
Run the Test Suite: http://pr715.butyr.org/test/

Commit:
[#1973] Make url input boxes only update if value changed

cadecairos commented:
/botio check

cadecairos commented:
/botio preview

mjschranz commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/16f1ce34b17df9d/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/af5b39d1cf3689e/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/16f1ce34b17df9d/output.txt

Total script time: 1.09 mins

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/af5b39d1cf3689e/output.txt

Total script time: 1.10 mins

Server is running on http://pr715.butyr.org/templates/basic/
Run the Test Suite: http://pr715.butyr.org/test/

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/89fdc1fe5f188ca/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/89fdc1fe5f188ca/output.txt

Total script time: 0.90 mins

Server is running on http://pr715.butyr.org/templates/basic/
Run the Test Suite: http://pr715.butyr.org/test/

jbuck commented on src/core/media.js:
```diff
@@ -7,8 +7,7 @@
-            "core/views/media-view"
+            "core/popcorn-wrapper"
           ],
           function( Logger, EventManagerWrapper, Track, PopcornWrapper, MediaView ){
```
If you're removing the media-view, remove the property for it too

cadecairos replied:
good catch

Commit:
[#1793]
1. Make Media header class update when editor opens and closes.
2. Remove old parameter for MediaView in src/core/media.js

jbuck commented:
Staged on master: https://github.com/mozilla/butter/commit/942740d6154707233aa9f5c14bc723de6f527b96
